pkg/services/pluginsintegration/clientmiddleware/caching_middleware.go

```diff
@@ -12,6 +13,8 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+var shouldCacheQuery = awsds.ShouldCacheQuery
```

Comment:
nit: I don't think we need to make this a global var, I think we can just call `awsds.ShouldCacheQuery` within the method below

Reply:
Never mind, I see this is because of the unit test. I recommend just leaving a comment here saying this is needed for testing

Reply:
done